dumpling/export/util.go

```diff
@@ -78,3 +78,27 @@ func string2Map(a, b []string) map[string]string {
 func needRepeatableRead(serverType version.ServerType, consistency string) bool {
 	return consistency != ConsistencyTypeSnapshot || serverType != version.ServerTypeTiDB
 }
+
+func infiniteChan[T any]() (chan<- T, <-chan T) {
+	in, out := make(chan T), make(chan T)
+
+	go func() {
+		var q []T
+		for {
+			e, ok := <-in
+			if !ok {
+				for _, e := range q {
+					out <- e
+				}
+				close(out)
+				return
+			}
+			q = append(q, e)
+			if len(out) == 0 && len(q) > 0 {
```

Comment:
`len(out)` is always 0

Reply:
the reader of `out`, which is data dumper goroutine, maybe busy and don't read tasks from `out` timely